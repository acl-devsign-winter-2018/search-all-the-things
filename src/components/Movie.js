import React, { Component } from 'react';
import PropTypes from 'prop-types';
// import styles from './movies.css';

export default class Movies extends Component {

  static propTypes = {
    movies: PropTypes.object.isRequired,
  };  

  render(){
    const { title, year, poster } = this.props.movies;
    
    return (
      <div>
        <ul>
          <li>
            <h2>{title}</h2>
            <h3>{year}</h3>
            <img src={poster}/>
          </li>
        </ul>
      </div>

    );

  }

}