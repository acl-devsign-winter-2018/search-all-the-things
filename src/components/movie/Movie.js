
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './movie.css';
import { Link } from 'react-router-dom';

export default class Movie extends Component {

  static propTypes = {
    imdbID: PropTypes.string,
    Poster: PropTypes.string,
    Title: PropTypes.string,
    Year: PropTypes.string,
  };

  render() {
    const { Title, Year, Poster, imdbID } = this.props;

    if(Poster === 'N/A') {
      return (
        <li className="posters">
          <Link className="missing-poster-a" to={`/movies/${imdbID}`}>
            <span className="missing-poster">?</span>
            <h2>{Title}</h2>
            <p>{Year}</p>
          </Link>
        </li>
      );
    }
    
    return (
      <li className="posters">
        <Link to={`/movies/${imdbID}`}>
          <img src={Poster}/>
          <h2>{Title}</h2>
          <p>{Year}</p>
        </Link>
      </li>
    );
  }
}