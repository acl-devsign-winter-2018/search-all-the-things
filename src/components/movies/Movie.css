@import '../../styles/variables.css';

.movie-item {
  display: flex;
  margin: 4rem 0 0;
  flex-direction: column;
}

.movie-description {
  text-align: left;
  margin: 0 1rem;
}

.poster {
  width: 300px;
  margin: 2rem;
}

.missing {
  width: 100%;
  height: 20%;
}

.missing-fill {
  display: block;
  background: $black;
  color: $white;
  font-size: 4rem;
  text-align: center;
}

@media screen and (min-width: 700px) {
  .movie-item {
    flex-direction: row;
  }
}