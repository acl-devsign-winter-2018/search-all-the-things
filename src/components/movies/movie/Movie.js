import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './movie.css';

export default class Movies extends Component {

  static propTypes = {
    movie: PropTypes.object.isRequired,
  };  

  render(){
    const { Title, Year, Poster } = this.props.movie;
    
    return (
      <div>        
        <li className="movie">
          <img src={Poster}/>
          <div className="movie-detail">
            <h2>{Title}</h2>
            <h3>{Year}</h3>
          </div>
        </li>       
      </div>
    );
  }
}