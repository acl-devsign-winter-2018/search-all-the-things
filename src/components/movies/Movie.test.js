import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import Movie from './Movie';

describe('Movie', () => {
  it('renders movie detail', () => {
    const Image = 'https://images-na.ssl-images-amazon.com/images/M/MV5BNzVlY2MwMjktM2E4OS00Y2Y3LWE3ZjctYzhkZGM3YzA1ZWM2XkEyXkFqcGdeQXVyNzkwMjQ5NzM@._V1_SX300.jpg';
    const wrapper = shallow(<Movie Title="Star Wars: Episode IV - A New Hope" Year="1997" Type="Movie" imdbID="tt0076759" Poster={Image}/>);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});