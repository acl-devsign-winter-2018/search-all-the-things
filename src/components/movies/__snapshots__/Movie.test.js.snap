// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie renders movie detail 1`] = `
<li>
  <Link
    replace={false}
    to="/movies/tt0076759"
  >
    <img
      src="https://images-na.ssl-images-amazon.com/images/M/MV5BNzVlY2MwMjktM2E4OS00Y2Y3LWE3ZjctYzhkZGM3YzA1ZWM2XkEyXkFqcGdeQXVyNzkwMjQ5NzM@._V1_SX300.jpg"
    />
    <h3>
      Star Wars: Episode IV - A New Hope
    </h3>
    <p>
      Year made: 
      1997
    </p>
    <p>
      Movie
       | IMDB ID: 
      tt0076759
    </p>
  </Link>
</li>
`;
